### PR TITLE
newrelic integration. Includes:

### PR DESCRIPTION
**Pull request by djcp**
* Install the newrelic configuration
* Inflect the names from the node name and a user supplied
 "environment_name"
* Inject newrelic into the matterhorn init script when it's enabled
* Configure newrelic logging

### CODE REVIEW

**Comment by djcp** [comment]
Assuming your stack is named "stack name", the nodes end up logging stuff to newrelic under "stack name" and "stack_name_admin1" (etc.) so that we get aggregate and individual node stats.

File: templates/default/newrelic.yml.erb [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+  class_transformer:
+    # This instrumentation reports the name of the user principal returned from 
+    # HttpServletRequest.getUserPrincipal() when servlets and filters are invoked.
+    com.newrelic.instrumentation.servlet-user:
+      enabled: false
+
+    com.newrelic.instrumentation.spring-aop-2:
+      enabled: false
+
+  # Phoebe: For pricing
```
Comment (gregorydlogan) [inline templates/default/newrelic.yml.erb]: Some clarity around what these keys are and what they do would be helpful.

File: templates/default/newrelic.yml.erb [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+  # Limits the number of lines to capture for each stack trace. 
+  # Default is 30
+  max_stack_trace_lines: 30
+
+  # Provides the ability to configure the attributes sent to New Relic. These
+  # attributes can be found in transaction traces, traced errors, Insight's 
+  # transaction events, and Insight's page views.
+  attributes:
+  
+    # When true, attributes will be sent to New Relic. The default is true.
```
Comment (gregorydlogan) [inline templates/default/newrelic.yml.erb]: This file appears to have a variety of different indentation levels, is that deliberate?

Comment (djcp) [inline templates/default/newrelic.yml.erb]: Yup! It's a YAML file.  The indentation denotes key nesting.

File: templates/default/newrelic.yml.erb (hunk 1 above)
Comment (djcp) [inline templates/default/newrelic.yml.erb]: it's all newrelic config specific, so it seems a little odd to document it here.

**Pull request closed without merging**